client/homebrew/pages/basePages/listPage/listPage.jsx

```diff
@@ -207,17 +210,18 @@ const ListPage = createClass({
 
 	renderBrewCollection : function(brewCollection){
 		if(brewCollection == []) return <div className='brewCollection'>
-			<h1>No Brews</h1>
+			<h1>{'No Brews'.translate(['userPage'])}.</h1>
 		</div>;
 		return _.map(brewCollection, (brewGroup, idx)=>{
 			return <div key={idx} className={`brewCollection ${brewGroup.class ?? ''}`}>
-				<h1 className={brewGroup.visible ? 'active' : 'inactive'} onClick={()=>{this.toggleBrewCollectionState(brewGroup.class);}}>{brewGroup.title || 'No Title'}</h1>
+				<h1 className={brewGroup.visible ? 'active' : 'inactive'} onClick={()=>{this.toggleBrewCollectionState(brewGroup.class);}}>{brewGroup.title || 'No Title'.translate()}</h1>
```

Comment:
Right now, this does not translate "Gazook89's Published Brews" / "Gazook89's Unpublished Brews".  This could be accomplished with a change on this line, and on the userPage.jsx.  This will change the `title` of the brewCollections, so someone wiser than I (@calculuschild , @G-Ambatte for example of a non-exhaustive list) might want to confirm this doesn't pose other dangers.  

But the gist of it is:

listPage line 217
```jsx
				<h1 className={brewGroup.visible ? 'active' : 'inactive'} onClick={()=>{this.toggleBrewCollectionState(brewGroup.class);}}>{`${brewGroup.title} ` + `${brewGroup.class} brews`.translate(['userPage']) || 'No Title'.translate(['userPage'])}</h1>

```

userPage line 34
```jsx
		const brewCollection = [
			{
				title : `${usernameWithS}`,
				class : 'published',
				brews : brews.published
			}
		];
```
result:
<img width="463" alt="image" src="https://github.com/naturalcrit/homebrewery/assets/58999374/228a4de5-a70d-48bd-b983-55e8d56abd20">


Probably there is some better way of concatenating the string together for the translate function, but right now I couldn't think of one.

Reply:
>> Right now, this does not translate "Gazook89's Published Brews" / "Gazook89's Unpublished Brews".
>
> [...] Probably there is some better way of concatenating the string together for the translate function, but right now I couldn't think of one.

I suggest the `translate()` function include an optional parameter of key:values to be merged into the translated string. such that the brewCollection title could be translated via 

`"username-published-brews".translate({"name": "Gazook89"})`

The english translation reference YAML file could then have for the `username-published-brews` key this translated value:

    ${name}'s published brews

or even this, to properly handle English possessive formation for names ending with "s":

    ${name}${name.endsWith('s') || name.endsWith('S') ? '’' : '’s'} published brews

This can handle simple substitution (e.g. inserting a name into the string), weird substitutions (e.g. irregular possessive s formation), multiple values (e.g. errorNumber, date, reference merged into a longer paragraph), and even basic math etc (e.g. `${number * factor}` or string operations (e.g. "le" or "la", according to circumstance).

And, of course, handle straight up static strings (e.g.`"Hello!": "Bon jour!"`)

----

In the `translate()` function itself, it would dynamically define an anon function that returns the filled-template string (via the `with (merge)` statement to get the merge values into the scope).

```js
String.prototype.translate = function (values) {
// exempli gratia..
// also ignoring the `groups` parameter functionality requirement
  const translationData = {
    "Names Published Brews": "${name}${name.endsWith('s') || name.endsWith('S') ? '’' : '’s'} Brassins Publiés",
    "Names Unpublished Brews": "${name}${name.endsWith('s') || name.endsWith('S') ? '’' : '’s'} Brassins Non Publiés",
    "Some string": "Jibber jabba",
    // Add more translations as needed
  };

  const key = Object.keys(values)[0];

  if (translationData[key]) {
    // Dynamically define the function
    const functionString = `with (values) return \`${translationData[key]}\`;`;
    const translationFunction = new Function('name', functionString);

    // Call the dynamically defined function
    return translationFunction(values[key]);
  } else {
    return `Translation not found for key: ${key}`;
  }
};
// CODE NOT TESTED
```

NB. this would require a refactor of code to date, since the current `translate()` function already has one optional parameter which is an unlabelled array of strings for scoping the translation lookup group). I suggest instead 

`"source-string".translate( { values: {"key": "value", "another-key": "another value"}, groups: ["array", "of", "strings"] })`

(Writing a function with an optional unlabelled parameter value is a code smell. Pass that thing inside an object with a key, please).

Reply:
OR, and just hear me out, simply change the way the userpage displays the titles, to have the username on top of the page, and then simply:

## Published brews

## Unpublished brews

and voilà, we just saved having to build that function at all.

Reply:
This still needs discussion.

Reply:
Collection titles is not the only place where we want to present a translated message that contains various values merged into it.

Reply:
I agree with both:  we can easily sidestep the issue on these particular headers without losing anything, but we may still benefit from the functionality that eric proposes.

Reply:
I agree with Eric's proposal. Languages are not always going to follow the same grammatical sequence, so we need a way for each language to place passed-in values in the correct point in the text.